.github/workflows/_finalize.yaml

```diff
@@ -8,22 +8,121 @@ on:
         description: Update the landing page badges with run outcomes
         default: false
         required: false
+      # upload badge
+      BADGE_FILES:
+        type: string
+        description: 'Name/bash regex for shields.io endpoint JSON files'
+        default: '*badge*.json'
+        required: false
+    outputs:
+      # upload badge
+      GIST_ID:
+        description: 'Id of the created Gist'
+        value: ${{ jobs.action.outputs.GIST_ID }}
 
 jobs:
-  upload-badge:
-    uses: ./.github/workflows/_upload_badge.yaml
-    secrets: inherit
-
-  report:
-    uses: ./.github/workflows/_summary.yaml
-    secrets: inherit
-
-  publish-badge:
-    needs: [upload-badge]
-    if: inputs.PUBLISH_BADGE == true
-    uses: ./.github/workflows/_copy_gist.yaml
-    with:
-      SRC_GIST_ID: ${{ needs.upload-badge.outputs.GIST_ID }}
-      DST_GIST_ID: ${{ vars.BADGE_ENDPOINT_GIST_ID }}
-      FILES: '.*badge.*.json'
-    secrets: inherit
+  action:
```

Comment:
Please use separate jobs for each of the reusable workflows that are being replaced.